src/index.cpp

```diff
@@ -28,8 +31,9 @@ namespace diskann
 // Initialize an index with metric m, load the data of type T with filename
 // (bin), and initialize max_points
 template <typename T, typename TagT, typename LabelT>
-Index<T, TagT, LabelT>::Index(const IndexConfig &index_config, std::unique_ptr<AbstractDataStore<T>> data_store,
-                              std::unique_ptr<AbstractGraphStore> graph_store)
+Index<T, TagT, LabelT>::Index(const IndexConfig &index_config, std::shared_ptr<AbstractDataStore<T>> data_store,
```

Comment:
I see the logic in `index_factory` to either assign initialize pq_data_store or assign it value of data_store. 

In case if we don't use PQ then we can just pass it as nullptr and keep everything as unique_ptr ?  or is there a case where pq_build = false and the code somehow uses pq_data_store ?

Reply:
we wanted to avoid the if(pq_build) conditions which were there in several places. this might make the code cleaner to read

Reply:
if we just set pq_data_store = data_store in case we dont have pq_build